### ISSUE TITLE
Use uri-bytestring instead of network-uri?

### ISSUE THREAD

**Issue by cdupont** [dustin/mqtt-hs#15]
Using uri-bytestring would be much better for constructing URIs. See for example: https://github.com/haskell-servant/servant/issues/139

https://hackage.haskell.org/package/uri-bytestring-0.1.5/docs/URI-ByteString.html

**Comment by dustin** [comment]
Hey, thanks for the pointer.  I hadn't heard of uri-bytestring, but I'll see if it makes my life better or worse.  :)

Is there usage of network-uri in here that causes you suffering as a user?

**Comment by cdupont** [comment]
Yeah, I wanted to construct a URI piece by piece using the constructors, and found it quite strange. For example, the port is defined as a String, that should include ":" (e.g. uriPort = ":42"). Not a big deal, just nit-picking :)
